x-pack/solutions/observability/plugins/streams/server/lib/streams/client.ts

```diff
@@ -306,6 +305,8 @@ export class StreamsClient {
     result: 'created' | 'updated';
     parentDefinition?: WiredStreamDefinition;
   }> {
+    await this.assertNoHierarchicalConflicts(definition.name);
```

Comment:
I think we should guard this so it's not called for unwired streams - `validateAndUpsertStream` takes a `StreamDefinition`, not only a `WiredStreamDefinition`, and `assertNoHierarchicalConflicts` calls `getAncestors` directly, so there is a risk that it's called with an unwired stream at some point which it never should (even if this is not the case yet).

Reply:
💯 I realized this as well when looking at the failing tests!

Reply:
(Hopefully, waiting for tests) fixed in https://github.com/elastic/kibana/pull/208914/commits/5c7bd81ed61fa86a9d42cba840c2d824e6e285a1